scripts/generator/graphql_generator/csharp/Cart.cs.erb

```diff
@@ -139,16 +154,35 @@ namespace <%= namespace %> {
                 CurrentCheckout = merger.Merge(CurrentCheckout, checkout);    
             }
 
+            if (userErrors.Count > 0) {
+                _UserErrors = userErrors;
+            } else {
+                _UserErrors = null;
+            }
+
             UpdateLineItemFromCheckout(CurrentCheckout);
+
+            return _UserErrors == null;
         }
 
         private void UpdateLineItemFromCheckout(Checkout checkout) {
-            // sometimes we may not query line items for instance when polling is being performed
-            try {
-                List<CheckoutLineItem> lineItems = (List<CheckoutLineItem>) checkout.lineItems();
+            if (checkout != null) {
```

Comment:
Nit: Question too - do we favor early returns vs adding another level of code? 

For example,

```csharp
if (checkout == null) {
    return;
}

... continue on your merry way!
```